src/stirling/source_connectors/perf_profiler/perf_profiler_bpf_test.cc

```diff
@@ -317,88 +314,52 @@ class PerfProfileBPFTest : public ::testing::TestWithParam<std::filesystem::path
   }
 
   void ConsumeRecords() {
-    const std::vector<TaggedRecordBatch> tablets = data_table_.ConsumeRecords();
-    ASSERT_NOT_EMPTY_AND_GET_RECORDS(columns_, tablets);
-    PopulateColumnPtrs(columns_);
+    constexpr uint32_t table_num = 0;
+    ASSERT_OK_AND_ASSIGN(columns_, source_.ConsumeRecords(table_num));
     auto target_row_idxs =
         FindRecordIdxMatchesPIDs(columns_, kStackTraceUPIDIdx, sub_processes_->pids());
 
     PopulateCumulativeSum(target_row_idxs);
     PopulateObservedStackTraces(target_row_idxs);
   }
 
-  void PopulateColumnPtrs(const types::ColumnWrapperRecordBatch& columns) {
-    trace_ids_column_ = columns[kStackTraceStackTraceIDIdx];
-    stack_traces_column_ = columns[kStackTraceStackTraceStrIdx];
-    counts_column_ = columns[kStackTraceCountIdx];
-    column_ptrs_populated_ = true;
-  }
-
-  void RefreshContext(const absl::flat_hash_set<md::UPID>& upids) {
-    absl::base_internal::SpinLockHolder lock(&perf_profiler_state_lock_);
-    ctx_ = std::make_unique<StandaloneContext>(upids);
-  }
-
-  void StartTransferDataThread() {
-    ASSERT_TRUE(source_ != nullptr);
-
-    sampling_period_ = source_->SamplingPeriod();
-    ASSERT_LT(sampling_period_, test_run_time_);
-
-    // Create an initial empty context for use by the transfer data thread.
-    const absl::flat_hash_set<md::UPID> empty_upid_set;
-    ctx_ = std::make_unique<StandaloneContext>(empty_upid_set);
-
-    transfer_data_thread_ = std::thread([this]() {
-      transfer_enable_ = true;
-      while (transfer_enable_) {
-        {
-          absl::base_internal::SpinLockHolder lock(&perf_profiler_state_lock_);
-          source_->TransferData(ctx_.get());
-        }
-        std::this_thread::sleep_for(sampling_period_);
-      }
-    });
-  }
-
-  void StopTransferDataThread() {
-    CHECK(transfer_data_thread_.joinable());
-    transfer_enable_ = false;
-    transfer_data_thread_.join();
-    source_->TransferData(ctx_.get());
+  Status InitAndStartSource(const bool fast_test) {
+    // These three commands need to run in sequence, i.e. because SetFastTestParameters() makes
+    // a call into the source connector freq. manager, it must be after init and before start.
+    PX_RETURN_IF_ERROR(source_.Init(sub_processes_->upids()));
+    if (fast_test) {
+      SetFastTestParameters();
+    }
+    PX_RETURN_IF_ERROR(source_.Start());
+    return Status::OK();
   }
 
-  std::chrono::duration<double> RunTest() {
+  Status RunTestCore() {
     const auto start_time = std::chrono::steady_clock::now();
 
     // While we sleep (here), the transfer data thread is running.
     // In that thread, TransferData() is invoked periodically.
     std::this_thread::sleep_for(test_run_time_);
-    StopTransferDataThread();
 
     // We return the amount of time that we ran the test; it will be used to compute
     // the observed sample rate and the expected number of samples.
```

Comment:
This comment is out of date (duration return value was removed).

On that note, I have a slight preference to keeping the assertion functions as taking a duration as an argument. While the current API this test class exposes makes it easy to write a new test, I find requiring an argument for a test assertion makes the API difficult to misuse (if you happened to call the elapsed time population and test assertion functions in the wrong order).

I'm ok with either option: removing this comment to make the new code consistent or keeping the previous test assertions / duration recording inputs. Just mentioning my slight preference this since it was related to the comment.

Reply:
Would rather keep as is. The "main" test cases are all self contained now. The symbolization artifacts removal test case and the out-of-context test case do not use this at all.